src/quark-core/src/QuarkScript.sol

```diff
@@ -118,4 +120,54 @@ abstract contract QuarkScript {
             sstore(key, value)
         }
     }
+
+    // Note: this may not be accurate after any nested calls from a script
+    function getActiveNonce() internal returns (bytes32) {
+        QuarkWallet self = QuarkWallet(payable(address(this)));
+
+        bytes32 activeNonceSlot = self.ACTIVE_NONCE_SLOT();
+        bytes32 value;
+        assembly {
+            value := sload(activeNonceSlot)
+        }
```

Comment:
do we not have `read()`/`write()` helpers that handle the `bytes32` case? would rather avoid `assembly` blocks and `sload`s if we can

Reply:
I think those are about to get isolation, so probably doesn't work?